kotest-property/src/commonMain/kotlin/io/kotest/property/arbitrary/edgecases.kt

```diff
@@ -0,0 +1,48 @@
+package io.kotest.property.arbitrary
+
+import io.kotest.property.Arb
+import io.kotest.property.EdgeConfig
+import io.kotest.property.RandomSource
+import io.kotest.property.Sample
+
+/**
+ * Randomly chooses an arb and then generates an edgecase from that arb.
+ * If the chosen arb has no edgecases, then another arb will be chosen.
+ * If all arbs have no edgecases, then returns null.
+ */
+fun <A> List<Arb<A>>.edgecase(rs: RandomSource): A? {
```

Comment:
Maybe mark this as tailrec and use when Sam just to indicate the terminal condition and recurs condition.

Reply:
Done